File: wiz-shop-java/src/main/java/com/example/wizshopjava/api/member/MemberService.java
package com.example.wizshopjava.api.member;

import org.springframework.stereotype.Service;

import com.example.wizshopjava.api.member.dto.MemberLoginRequest;
import com.example.wizshopjava.api.member.dto.MemberRegisterRequest;
import com.example.wizshopjava.common.helper.JwtHelper;
import com.example.wizshopjava.domain.member.repository.MemberRepository;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class MemberService {

	private final MemberRepository memberRepository;
	private final JwtHelper jwtHelper;

	public void register(MemberRegisterRequest req) {
		if (memberRepository.existsByEmail(req.email())) {
			throw new RuntimeException("이미 가입되어있는 이메일입니다!");
		}
		memberRepository.save(req.toEntity());
	}

	public String login(MemberLoginRequest req) {
		return memberRepository.findByEmail(req.email())
			.filter(member -> member.getPassword().equals(req.password()))
			.map(jwtHelper::generateAccessToken)
			.orElseThrow(() -> new RuntimeException("잘못된 아이디 혹은 패스워드입니다."));
	}
}

File: wiz-todo/src/main/kotlin/com/example/wiztodo/api/service/MemberService.kt
package com.example.wiztodo.api.service

import com.example.wiztodo.api.controller.dto.request.MemberRegisterRequest
import com.example.wiztodo.common.JwtHelper
import com.example.wiztodo.domain.member.entity.Member
import com.example.wiztodo.domain.member.repository.MemberRepository
import org.springframework.stereotype.Service

@Service
class MemberService(
    private val memberRepository: MemberRepository,
    private val jwtHelper: JwtHelper
) {

    fun register(req: MemberRegisterRequest) {
        /**
         * 2. 위 정보중 이메일로 가입된 사용자가 이미 존재하는지 확인한다.
         * 3. 위 정보중 패스워드를 암호화한다.
         * 4. 사용자 정보를 생성해 데이터베이스에 저장한다.
         */
        if (memberRepository.existsByEmail(req.email)) {
            throw RuntimeException()
        }
        // ...
        val member = Member(email = req.email, password = req.password)
        memberRepository.save(member)
    }

    fun login(req: MemberRegisterRequest): String {
        val member = memberRepository.findByEmail(req.email)
            ?: throw RuntimeException()
        if (member.validPassword(req.password)) {
            throw RuntimeException()
        }
        return jwtHelper.generateAccessToken(member.id!!)
    }
}